libraries/AP_BattMonitor/AP_BattMonitor_UAVCAN.cpp

```diff
@@ -65,6 +65,9 @@ AP_BattMonitor_UAVCAN* AP_BattMonitor_UAVCAN::get_uavcan_backend(AP_UAVCAN* ap_u
             AP::battery().get_type(i) == AP_BattMonitor_Params::BattMonitor_TYPE_UAVCAN_BatteryInfo) {
 
             AP_BattMonitor_UAVCAN* batmon = (AP_BattMonitor_UAVCAN*)AP::battery().drivers[i];
+            if(batmon->_ap_uavcan != nullptr || batmon->_node_id != 0) {
```

Comment:
I'm unsure why you are checking the `nullptr` here? It's already been check to not be `nullptr` before we get here, and unless I'm reading this wrong (quite possible, I've just woken up), doesn't this result in never finding an empty driver?

Reply:
@WickedShell  If it is not empty, the uavcan driver has been registered. You need to find an empty uavcan driver.

Reply:
@WickedShell Hello, can this pass?